### ISSUE TITLE
Unusual yoda expressions

### ISSUE THREAD

**Issue by dcousens** [feross/standard#91]
I recently had a case where I was using the following code in a test:

``` javascript
{
  a: true,
  b: false,
  c: 1 > 2
}
```

Standard blew up on me with `Expected literal to be on the right side of >`.
Apparently this is due to it attempting to adhere a yoda condition ruling.

Is this intentional?

I understand this is most likely a question for eslint, not standard, but I thought I'd ask first.

**Comment by feross** [comment]
It seems that `eslint` considers any expression where the left side is a constant to be a "yoda expression". That seems okay behavior to me, though the error is a bit misleading in this case.

**Comment by dcousens** [comment]
So @feross we think its acceptable to not allow constant expressions?  That's ridiculous no?  Its a perfectly valid case of something you might want?

**Comment by feross** [comment]
@dcousens Constant expressions are still allowed, of course :-) For example: `var x = 'my string'` and `var y = 42`.

One side effect of the "no yoda expressions" rule is that comparison expressions that always evaluate to the same thing (like `1 > 2`) should just be inlined. Is there a reason why you can't change the test code to:

```json
{
  a: true,
  b: false,
  c: false
}
```

**Comment by feross** [comment]
Btw, I don't actually think that preventing constant comparisons like `1 > 2` is very important - happy to turn that off if there's a way to do so.

If anything, `eslint` should make it a separate rule from the yoda expressions one, which is a bit more useful. Have you filed a bug on `eslint`?

**Comment by dcousens** [comment]
@feross I personally don't see any difference between: `2 < 1` and `2 + 1`.  In my mind they are both constant expressions, and yet the former is being picked up as erroneous.

I was thinking of pushing this upstream to eslint, just figured I'd ask here first.

**Comment by feross** [comment]
Ah, you're right. If `1 + 2` is allowed, then `2 < 1` should be too. This is an eslint bug, in my opinion.

**Comment by dcousens** [comment]
@feross did we file this in `eslint`?

**Comment by feross** [comment]
@dcousens I don't remember filing this on eslint.

**Comment by feross** [comment]
@dcousens Did you ever file this bug on eslint?

**Comment by dcousens** [comment]
@feross I haven't yet, no.

**Comment by feross** [comment]
Bug filed on eslint: https://github.com/eslint/eslint/issues/2867

**Comment by feross** [comment]
Bug fixed on eslint master.

**Comment by feross** [comment]
This is fixed in eslint 0.24.1. Run `npm update` to get the fix.
